port/serial_port.cpp

```diff
@@ -0,0 +1,181 @@
+/* Copyright 2020-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <thread>
+#include <chrono>
+#include <memory>
+#include "serial/serial.h"
+
+#include "serial_port.h"
+
+using std::chrono::high_resolution_clock;
+using std::chrono::duration_cast;
+using std::chrono::duration;
+using std::chrono::milliseconds;
+using namespace std::chrono_literals;
+
+loader_serial_config_t serial_config;
+std::unique_ptr<serial::Serial> serial_port;
```

Comment:
As far as I see, we're not really using the properties of the unique pointer here, this being a static variable. For deinit, it would make no difference if we were to delete the pointer or just directly delete the object, but I might be wrong here.

Reply:
Wasn't sure about cleanup. The unique ptr would do that by itself. A propper deinit call would be fine too

Reply:
True, this way when main returns the destructor is called and the OS housekeeping related to the port is done. Making the object itself static is not possible as we don't yet have the config for constructing it. Best leave it as is, but provide the option of manual deinit as well.